### ISSUE TITLE
Select / filter and previously selected options behavior

### ISSUE THREAD

**Issue by RafaPolit** [huridocs/uwazi#2922]
What should be the correct behavior on the selects, multiselects, and filters when there are values previously selected?

## Scenario 1

- You have a filter to limit the country cases belong to
- You have selected already Colombia
- You want to include its neighbors Ecuador and Venezuela, therefore type `Ecua` in the filter

What is the expected behavior?
A.  You now see countries that match `Ecua` and nothing more? (this is the current behavior)
B.  You see countries that match `Ecua` but somehow you also see `Colombia` is still selected (even if Colombia does not match at all the `Ecua` search)

I have the feeling that a form to achieve scenario B would be ideal, but I understand the argument that Colombia does not match at all your current `Ecua` filter.  

While, in this case, B is probably the right answer, try to think about this on more extreme scenarios...

## Scenario 2

- You have already assigned 20 values to the `keyword` property of a Law
- You want to add another keyword and look for it in the search, limiting by name
- If aiming for behavior B above, how would you display all 20 previous values, but still make it so that filtering by the new value is useful and returns the values you were looking to filter by?

cc @simonfossom @txau

**Comment by txau** [comment]
@RafaPolit for scenario 1 the criteria is based in geolocation proximity? I don't think we should even be thinking about this approaches for now.

For scenario 2, if I'm understanding correctly the problem is that when you search for the different options in filter you loss sight of the currently selected filters. I guess this is already addressed in @simonfossom 's proposal https://github.com/huridocs/uwazi/issues/2609

**Comment by RafaPolit** [comment]
No, nothing to do with proximity.  Replace the countries with A, B, C.  It's just a question of showing already selected values and new search options. Yes, @simonfossom proposal partially solves this, still there are more edge cases we could discuss.  But yes, we can do that on the other issue.

**Comment by simonfossom** [comment]
@RafaPolit you could solve it by grouping search of a filter with unselected options. In other words, put selected on top. This way selected options are not part of the search. 

What do you think?  🤔 

![Frame 1164](https://user-images.githubusercontent.com/8134850/83867390-a4498200-a729-11ea-9471-05023a892748.png)

FYI put together all the edge cases and let's have a call about it.

**Comment by RafaPolit** [comment]
I actually like a lot the end result of that.  My concern are things that start 'changing positions', so... imagine a long list, and you have Colombia up there... you will pull your hairs out trying to find Colombia in its alphabetical position below, just to figure out it was 'pulled' from the group to another side.

But yes, that is the general idea.  I agree, lets put all the problematics together and fix them all at once.

**Comment by simonfossom** [comment]
I get your point. 🤔  

It's usually solved with micro-interactions.
[This example](https://i.pinimg.com/originals/35/c0/14/35c014eb47da5e016582c50be65cf730.gif) is too playful but it demonstrates the point. 

---
Ok, I've played around a bit. Here's a [prototype of the animation](https://www.figma.com/proto/Zw9JkOl11WCgdFfIKugCGy/Uwazi-connections?node-id=478%3A11&viewport=-8340%2C2229%2C0.6031964421272278&scaling=min-zoom)

**Comment by txau** [comment]
related #4155
